newapi/tests/conftest.py

```diff
@@ -53,7 +60,121 @@ def client(app):
 def pytest_addoption(parser):
     parser.addoption("--ghpr", action="store_true", help="enable GitHub integ tests")
     parser.addoption("--proddb", action="store_true", help="uses data from prod DB")
+    parser.addoption("--create-db", action="store_true", help="populate a dedicated DB")
+
 
 def pytest_configure(config):
     pytest.run_ghpr = config.getoption("--ghpr")
     pytest.proddb = config.getoption("--proddb")
+    assert pytest.proddb is False, "--proddb is disabled"
+    pytest.create_db = config.getoption("--create-db")
+
+
+def sudopg(cmd, check=True):
+    cmd = ["/usr/bin/sudo", "-u", "postgres", "psql", "-c", cmd]
+    print(cmd)
+    out = subprocess.run(cmd, check=check, stdout=PIPE, stderr=PIPE).stdout
+    out = out.decode().strip()
+    if out:
+        print(out)
+
+
+@pytest.fixture(scope="session")
+def setup_database_part_1():
+    # Create database and users.
+    # Drop and recreate database if exists.
+    if not pytest.create_db:
+        return
+
+    if os.path.exists("/usr/bin/sudo"):
+        print("Creating PostgreSQL user and database")
+        sudopg("DROP DATABASE IF EXISTS oonitestdb", check=True)
+        sudopg("DROP ROLE IF EXISTS oonitest", check=True)
+        sudopg("CREATE USER oonitest WITH ENCRYPTED PASSWORD 'test'", check=False)
+        sudopg("CREATE DATABASE oonitestdb WITH OWNER 'oonitest'", check=False)
+        sudopg("GRANT ALL PRIVILEGES ON DATABASE oonitestdb TO oonitest")
+
+    else:
+        # On github sudo is missing and the database is already created
+        print("Sudo not found - not creating PostgreSQL database")
+
+
+@pytest.fixture(scope="session")
+def checkout_pipeline(tmpdir_factory):
+    if not pytest.create_db:
+        return
+    d = tmpdir_factory.mktemp("pipeline")
+    if d.isdir():
+        shutil.rmtree(d)
+    cmd = f"git clone --depth 1 https://github.com/ooni/pipeline -q {d}"
+    print(cmd)
+    cmd = cmd.split()
+    subprocess.run(cmd, check=True, stdout=PIPE, stderr=PIPE).stdout
+    return Path(d)
+
+
+def run_sql_scripts(app):
+    log = app.logger
+    # for i in ["1_metadb_users.sql", "2_metadb_schema.sql", "3_test_fixtures.sql"]:
+    query = ""
+    for i in ["2_metadb_schema.sql", "3_test_fixtures.sql"]:
```

Comment:
These are already run as part of the docker-compose db definition. If you decide to switch to the pattern of running them in here, you probably should remove them from: https://github.com/ooni/api/pull/258/files#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3R12

Reply:
See previous comment.